### PR TITLE
Allow custom service code to modify/add the response HTTP headers

### PR DESCRIPTION
**Pull request by vdantu**
Before or while filing an issue please feel free to join our [ slack channel](https://join.slack.com/t/mms-awslabs/shared_invite/enQtNDk4MTgzNDc5NzE4LTBkYTAwMjBjMTVmZTdkODRmYTZkNjdjZGYxZDI0ODhiZDdlM2Y0ZGJiZTczMGY3Njc4MmM3OTQ0OWI2ZDMyNGQ) to get in touch with development team, ask questions, find out what's cooking and more!

## Issue #, if available:
PR to allow custom service code to add/modify http response headers.

## Testing done:

**To run CI tests on your changes refer [README.md](https://github.com/awslabs/mxnet-model-server/blob/master/ci/README.md)**

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: frontend/server/src/main/java/com/amazonaws/ml/mms/util/codec/ModelResponseDecoder.java [hunk 1]
```diff
@@ -61,6 +61,13 @@ protected void decode(ChannelHandlerContext ctx, ByteBuf in, List<Object> out) {
                 }
                 prediction.setContentType(CodecUtils.readString(in, len));
 
+                len = CodecUtils.readLength(in, maxBufferSize);
+                if (len == CodecUtils.BUFFER_UNDER_RUN) {
+                    return;
+                }
+
+                prediction.setHeaders(CodecUtils.readMap(in, len));
```
Comment (frankfliu) [inline frontend/server/src/main/java/com/amazonaws/ml/mms/util/codec/ModelResponseDecoder.java]: I don't think it's good idea mix json and binary encoding together.

File: frontend/server/src/main/java/com/amazonaws/ml/mms/util/messages/Predictions.java [hunk 2]
```diff
@@ -12,12 +12,23 @@
  */
 package com.amazonaws.ml.mms.util.messages;
 
+import java.util.Map;
+
 public class Predictions {
 
     private String requestId;
     private String contentType;
+    Map<String, String> headers;
```
Comment (vrakesh) [inline frontend/server/src/main/java/com/amazonaws/ml/mms/util/messages/Predictions.java]: nitpick, declare this as private explicitly as well

File: frontend/server/src/main/java/com/amazonaws/ml/mms/util/codec/CodecUtils.java [hunk 3]
```diff
@@ -12,9 +12,13 @@
  */
 package com.amazonaws.ml.mms.util.codec;
 
+import com.google.gson.Gson;
```
Comment (vrakesh) [inline frontend/server/src/main/java/com/amazonaws/ml/mms/util/codec/CodecUtils.java]: We should use GSON object in utils.JsonUtils instead

**Comment by vdantu** [comment]
@frankfliu, @vrakesh : Reviews will be greatly appreciated :)

**Pull request merged**